File: 9_busqueda_binaria.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int busquedaBinaria(vector<int> arr, int valor){
    sort(arr.begin(), arr.end());
    int izq=0, der=arr.size()-1;
    while(izq<=der){
        int mid=(izq+der)/2;
        if(arr[mid]==valor) return mid;
        else if(arr[mid]<valor) izq=mid+1;
        else der=mid-1;
    }
    return -1;
}
int main(){
    vector<int> arr = {10,20,30,40,50};
    cout << busquedaBinaria(arr, 40);
}

File: 9_busqueda_binaria.java
import java.util.*;
public class Main {
    static int binaria(int[] arr, int val){
        Arrays.sort(arr);
        int l=0,r=arr.length-1;
        while(l<=r){
            int m=(l+r)/2;
            if(arr[m]==val) return m;
            else if(arr[m]<val) l=m+1;
            else r=m-1;
        }
        return -1;
    }
    public static void main(String[] args){
        System.out.println(binaria(new int[]{10,20,30,40,50},40));
    }
}